avm/res/sql/server/key/main.bicep

```diff
@@ -0,0 +1,47 @@
+metadata name = 'Azure SQL Server Keys'
+metadata description = 'This module deploys an Azure SQL Server Key.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the key. Must follow the [<keyVaultName>_<keyName>_<keyVersion>] pattern.')
+param name string
```

Comment:
This should either be optional or required, not both. As you can see in line 32, we actually have a default. Either we remove that, or we change the behavior of this parameter

Reply:
Done: I made the name parameter optional (nullable) and converted one of the tests to deploy keys with a default name. See: https://github.com/Azure/bicep-registry-modules/pull/637/commits/4c53d6ce1d4db91ca24958e5311e7b09c43ed128 and https://github.com/Azure/bicep-registry-modules/pull/637/commits/3241ddb6769846178ace5ab1485edd344d832ca6